framework/grid/ActionColumn.php

```diff
@@ -139,7 +150,7 @@ protected function initDefaultButtons()
                     'title' => Yii::t('yii', 'View'),
                     'aria-label' => Yii::t('yii', 'View'),
                     'data-pjax' => '0',
-                ], $this->buttonOptions);
+                ], isset($this->buttonOptions['view']) ? $this->buttonOptions['view'] : $this->buttonOptions);
```

Comment:
--disclaimer: I'm not in the Yii team --

I'd suggest introducing a different variable for default options `$defaultButtonOptions` and always merging those into whatever button configuration is passed via `$buttonOptions`. That way it's clear what the difference is.